### DISCUSSION TITLE
Various Rotation Issues

### DISCUSSION THREAD

**Discussion by wppronto** [AzuraCast/AzuraCast#7025] [Q&A]
I'm trying to use AzuraCast's playlist weights to set up a decent rotation. I'd like to avoid using custom Liquidsoap code because I don't want to boot listeners if I ever modify it. (I know that isn't supposed to happen when reloading the configuration, but it seems to.) Plus, I know that a lot of work has been put into AzuraCast's native playlist mechanisms, so I'd like to utilize them. But, in testing, I see a few issues.

For test purposes, I'd like to play about 8 1980s songs per hour, 6 1970s songs, and 2 1990s. And, of course, I'd like them to rotate pretty evenly without staying on one playlist too long. I also have sweepers set for every 3 songs. For testing, I also set the repeat prevention number very low. I set the playlist weights to the approximate number of songs I wanted per hour:

1980s: Playlist weight 8.
1970s: Playlist weight 6.
1990s: Playlist weight 2.

If I analyze a couple of hours of programming, here are the playlists that get pulled from (in chronological order);

Sweepers
Sweepers
Cue test
1970s
1980s
1970s
Sweepers
1990s
1990s
1990s
1970s
Sweepers
1990s
1970s
1970s
Sweepers
1980s
1980s
1980s
1980s
Sweepers
1990s
1980s
1990s
1980s
Sweepers
1970s
1980s
1970s
Sweepers
1980s
1980s
1970s
Sweepers
1970s
1970s
1980s
Sweepers
1990s
1980s
1980s
1980s

Issue #1: At the beginning, you can see that two sweepers played in a row. I'm assuming I can prevent this by choosing "only play one item from this playlist at the scheduled time" (or however it's worded). But is this a bug, since sweepers are set to play once every 3 songs?

Issue #2: Sweepers are set to play every 3 songs, but sometimes it's 3, sometimes it's 4.

Issue #3: Near the beginning, you can see that a song played from the "Cue Test" playlist. However, that playlist is disabled...hasn't been active for weeks.

Issue #4: The rotation is very uneven. For example, at one point there are 3 1990s songs in a row. At another point there are 4 1980s songs in a row. At multiple other points there are 3 songs from one playlist in a row. So, even though 1970s songs are supposed to be a large part of the rotation, at one point there 8 songs (about half an hour) without a 1970s song. But later, there are 3 1970s songs in a row.

How can I resolve these items? I know some of it could be user error...just not sure which parts.

**Comment by Moonbase59** [comment]
Re \#3: I have seen the same (Christmas playlist showing in mid-March!). It seems that if you have the _same song_ in _multiple playlists_, a "random" (first? last?) playlist name is shown, out of the playlists that the song is in.

**Comment by wppronto** [comment]
Thank you! That makes sense.

**Comment by wppronto** [comment]
Actually, I just saw that disabled playlist appear again. I checked the song in question, and it ONLY appears on that playlist. So a disabled playlist is still being played from time to time, even if it doesn't contain a song that's also on another playlist.

I don't know how to solve the other issues either...sweepers played twice, sweepers sometimes playing every 4 songs even though they're set for every 3 songs, etc.

No sure if my system is acting up, or if this is just how it is.

**Comment by wppronto** [comment]
This is very frustrating, as the playlist settings control the sound of the station, but several things aren't working. If it's user error, I'd like to figure out what I'm doing wrong. If it's bugs, I'm happy to help troubleshoot.

I guess I should ask, is anyone else having issues like these?
